### PR TITLE
MBM-418 update armor energy & MBM-164 fix structure generation

### PR DESCRIPTION
**Pull request by insertStringHere**

### CODE REVIEW

File: src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java
Comment (MattKirchh) [inline src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java:1]: Could you explain in layman's terms how this new armor recovery logic works? I don't quite understand it.

Comment (MattKirchh) [reply src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java:1]: I think you just need to be checking `getAvailableManaCap()` instead of `getTotalManaCap()` on line 297.

Comment (insertStringHere) [reply src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java:1]: Done

**Review by MattKirchh — APPROVED** [review]

**Review by YonatanWN — APPROVED** [review]
